src/v2/cookbook/vue-route-change-handling.md

```diff
@@ -0,0 +1,376 @@
+  created() {
+    this.loadData();
+  },
+  methods: {
+    loadData() { /* logic to load data */ },
+  },
+});
+```
+
+```html
+<router-view :key="$route.fullPath"></router-view>
+```
+
+This will assure that when going from `/orders` to `/orders?page=2` or from `/orders?page=2` to `/orders?page=1`, a fresh instance of `OrderIndex` component is created, triggering the `created` hook and downloading fresh data. There's also a subtle alternative to it:
+
+```html
+<router-view :key="$route.path"></router-view>
+```
+
+The main difference is that changes on the query eg: going from `/orders` to `/orders?page=2` won't make up a different `key` and therefore, no data is fetched. However going to a different route will keep the same behaviour. 
+
+Sometimes we don't want to recreate the view component so we go with key-less router-view:
+
+```html
+<router-view></router-view>
+```
+
+and `$watch` the `$route` object:
+
+```js
+Vue.component('OrderIndex', {
+  watch: {
+    $route: {
+      immediate: true,
+      handler: 'loadData',
+    },
+  },
+  methods: {
+    loadData() { /* logic to load data */ },
+  },
+});
+```
+
+<p class="tip">Note! [VueRouter](https://router.vuejs.org/en/) supports [named views](https://router.vuejs.org/en/essentials/named-views.html) and [nested routes](https://router.vuejs.org/en/essentials/nested-routes.html) so the component behaviour depends on the assigned `key` to each `router-view`. It's best to avoid mixing different `key` expressions.</p>
+
+Now each time the route changes, `OrderIndex` reloads its data.
+
+<p class="tip">Remember! Don't modify `$route.query` directly - it does not change the url, use `$router.push` or `$router.replace` instead.</p>
+
+## Simple, but not good enough
+While both `:key` methods work, there are some drawbacks to each one. Lets start with:
+
+```html
+<router-view :key="$route.fullPath"></router-view>
+```
+
+The `$route.fullPath` returns the full url including the query e.g. `/orders?page=2&filter=search`, so what happens when you visit from `/orders` to `/orders?page=1`? The page is recreated with new instances of components and the data is reloaded, yet the route is technically the same as before. At this point the only possible workaround would be to cache the results using something like [VueX](https://vuex.vuejs.org) or plain object. We could use `$route.path` as a `key`, but this will require introducing a watcher and handling query changes which sometimes can be difficult and unexpected side effects if implemented incorrectly. There are more examples why `:key` on the `router-view` could introduce more reloads than necessary which we'll cover further.
+
+Next we have the watch `$route` path without `key` on `router-view`. This gives us more freedom when to reload the data, but it also mean we have to handle it manually. Taking the `/orders` to `/orders?page=1` example, you now can decide if the page query parameter really changed and when to reload the data:
```

Comment:
I don't completely get the first 2 sentences.
What are you trying to explain here?
It's also easier to watch `$route.query` directly

Reply:
First two sentences are basically saying - we're going for the good old `<router-view></router-view>` logic, no keys etc.. will tweak it a bit.

I kind of think watching `$route.query` or `$route` is just preference and situation dependent. If you have a modal in the same page but it has it's own route, then you might want to watch `$route` to check the from/to so that closing the modal does not start reloading the data etc. though `$route.query` might be more used. I can change this if you prefer it.